src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1030,6 +1030,7 @@ public function withCount($relations)
                 $relation->getRelated()->newQuery(), $this
             );
 
+            $this->mergeModelDefinedRelationWheresToHasQuery($query, $relation);
             call_user_func($constraints, $query);
```

Comment:
@barryvdh I think that ``call_user_func`` should come before ``mergeModelDefinedRelationWheresToHasQuery``.

That's because you can potentially remove global scopes from the ``$query`` in the callback and ``mergeModelDefinedRelationWheresToHasQuery` takes care of removing them from the ``$relation`` too.